From 581863dc41abbd7a2c83a3980de47476bda958a9 Mon Sep 17 00:00:00 2001
From: althafvly <althafvly@gmail.com>
Date: Sat, 30 Apr 2022 16:02:32 +0530
Subject: [PATCH] tools: Add script for dynamically adding fonts

- This tool used to combined two fonts_customization xmls

eg: python3 fonts_combiner.py font1.xml font2.xml combined_font.xml
Change-Id: I22b4685b7c8db011d1b559892e1276a82bf8dfbd
---

diff --git a/tools/fonts_combiner.py b/tools/fonts_combiner.py
new file mode 100644
index 0000000..2d7df03
--- /dev/null
+++ b/tools/fonts_combiner.py
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+import sys
+import os
+from xml.etree import ElementTree
+import xml.dom.minidom as md
+
+
+def combine(files):
+    if not len(files) == 2:
+        raise ValueError("Wrong number of arguments %s" % len(files))
+
+    first = None
+    for filename in files:
+        data = ElementTree.parse(filename).getroot()
+        if first is None:
+            first = data
+        else:
+            first.extend(data)
+    if first is not None:
+        return (ElementTree.tostring(first)).decode()
+
+def pretty(xml):
+    dom = md.parse(xml)
+    pretty_xml = dom.toprettyxml()
+    pretty_xml = os.linesep.join([s for s in pretty_xml.splitlines()
+                                  if s.strip()])
+    with open(xml, "w") as file1:
+        file1.writelines(pretty_xml)
+
+if __name__ == "__main__":
+    # First 2 args are files to combine.
+    combined_xml = combine(sys.argv[1:3])
+    # output file
+    outfile = sys.argv[3]
+    with open(outfile, "w") as f:
+        f.writelines(combined_xml)
+    pretty(outfile)
